xayn-ai/src/coi/point.rs

```diff
@@ -47,65 +55,64 @@ impl From<PositiveCoi_v0_1_0> for PositiveCoi {
     }
 }
 
-#[cfg(test)]
-impl PositiveCoi_v0_0_0 {
-    pub fn new(id: CoiId, point: Embedding) -> Self {
-        Self {
-            id,
-            point,
-            alpha: 1.,
-            beta: 1.,
-        }
-    }
-}
-
-#[cfg(test)]
-impl PositiveCoi_v0_1_0 {
-    pub fn new(id: CoiId, point: Embedding) -> Self {
+impl From<PositiveCoi_v0_2_0> for PositiveCoi {
+    fn from(coi: PositiveCoi_v0_2_0) -> Self {
         Self {
-            id,
-            point,
-            alpha: 1.,
-            beta: 1.,
+            id: coi.id,
+            point: coi.point,
+            view_count: 1,
+            view_time: Duration::default(),
+            last_time: SystemTime::UNIX_EPOCH,
         }
     }
 }
 
-impl PositiveCoi {
-    pub fn new(id: CoiId, point: Embedding) -> Self {
-        Self { id, point }
-    }
-}
-
 #[derive(Clone, Deserialize, Serialize)]
 #[cfg_attr(test, derive(Debug, PartialEq))]
 pub(crate) struct NegativeCoi {
     pub id: CoiId,
     pub point: Embedding,
 }
 
-impl NegativeCoi {
-    pub fn new(id: CoiId, point: Embedding) -> Self {
-        Self { id, point }
-    }
-}
-
 pub(crate) trait CoiPoint {
-    fn new(id: CoiId, embedding: Embedding) -> Self;
+    fn new(id: CoiId, point: Embedding, viewed: Option<Duration>) -> Self;
     fn id(&self) -> CoiId;
     fn set_id(&mut self, id: CoiId);
     fn point(&self) -> &Embedding;
     fn set_point(&mut self, embedding: Embedding);
+    fn update_view(&mut self, viewed: Option<Duration>) {
+        #![allow(unused_variables)]
+    }
```

Comment:
Why not this?
```suggestion
    fn update_view(&mut self, _viewed: Option<Duration>) {}
```

Reply:
because then rust-analyzer shows `_viewed` as the field name hint also in the case where you provide a custom implementation, which might be confusing as it looks like this field would always be unused, but actually it's only unused in the default implementation.